text/src/autogluon/text/automm/predictor.py

```diff
@@ -566,21 +566,65 @@ def _fit(
 
         if trainer.global_rank == 0:
             top_k_avg_ckpt_path = os.path.join(save_path, "model.ckpt")
+            # Obtain the best K models sorted from the best performance to the worst performance
+            if minmax_mode == 'min':
+                best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
+                                              key=lambda ele: ele[1])
+                best_k_models_path = [ele[0] for ele in best_k_models_sorted]
+            elif minmax_mode == 'max':
+                best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
+                                              key=lambda ele: ele[1],
+                                              reverse=True)
+                best_k_models_path = [ele[0] for ele in best_k_models_sorted]
```

Comment:
This kind of logic is pretty prone to bugs, any thoughts on standardizing to always be lower=better regardless of metric to simplify downstream code? This is done in Tabular and vastly simplified a lot of logic + reduced bugs

Reply:
In fact, I feel that we may use `minmax_mode` or `mode`, which will support "max" and "min". This is the design adopted in packages such as PT Lightning: https://pytorch-lightning.readthedocs.io/en/stable/common/early_stopping.html and Ray Tune: https://docs.ray.io/en/latest/tune/api_docs/suggestion.html . This makes us consistent to the other packages.